### ISSUE TITLE
[Feature request] Night mode/dark mode

### ISSUE THREAD

**Issue by lappet** [olivierkes/manuskript#388]
A lot of editors and websites these days come with a night mode or a dark mode where the background text is not white in color. Can we get that please?

**Comment by RiderExMachina** [comment]
As a lover of dark themes I agree!

**Comment by gedakc** [comment]
Did you know that Manuskript already supports manually changing item colours?

To search through the existing issues raised for Manuskript you can add and subtract terms.  The steps to search all issues for the word "**dark**" are:

1. Click on **Issues** in the [Manuskript git page](https://github.com/olivierkes/manuskript/issues).
2. Remove "`is:open`" from the **Filters**.
3. Add "`dark`".
4. Press **Enter** key.

This should create the following [list of issues which contain the word "**dark**"](https://github.com/olivierkes/manuskript/issues?utf8=%E2%9C%93&q=is%3Aissue+dark).

See [issue 183 - Index card text almost invisible in dark themes](https://github.com/olivierkes/manuskript/issues/183) for an example of how some users have configured dark themes.

Does that answer your question?

**Comment by HorsaTheHetrisiniLord** [comment]
No actually... I don't see exactly where it says how to configure a dark theme in that issue...

I'd much rather have an official dark theme instead of manually having to configure it.

**Comment by gedakc** [comment]
I believe the dark theme in the issue #183 mentioned above was selected at the desktop level.

For example in Kubuntu 16.04 I can select "**Breeze Dark**" under **K -> Settings -> System Settings -> Workspace Theme -> Look And Feel**.

![kde-workspace-theme](https://user-images.githubusercontent.com/10405019/46682623-473fa680-cbab-11e8-9914-20415cd05f35.png)

Manukript then uses the chosen workspace theme.

You can further refine the text editor colours using Manuskript's **Edit -> Settings-> Views -> Text editor**.

![manuskript-text-editor-settings](https://user-images.githubusercontent.com/10405019/46682717-8d950580-cbab-11e8-9b3f-8f8e7c701e40.png)

**Comment by gedakc** [comment]
Comment from AlucardNoir copied from [Manuskript 0.7.0 web site page](http://www.theologeek.ch/manuskript/2018/08/15/manuskript-0-7-0/):

I'm posting this here because I don't want to make a GitHub account and have yet another account and password to bog me down. In regards to the requested Dark Mode/Theme( https://github.com/olivierkes/manuskript/issues/388 ) while things might work in Linux like that, on Windows things are a lot different. Dark Mode was a joke before the last update and frankly still is somewhat of a joke even after the October update. For windows, at least at the present, the program must supply it's own dark theme. Hell, even MS Office doesn't use a windows flag to turn on the inbuilt dark themes it has, and that's an MS product - MS is bad at doing dark mode.

**Comment by bobf32** [comment]
+1 for easy way to switch to dark mode.

Loving the product so far, great work.

**Comment by zja203** [comment]
The thing is, the color-changing options don't seem to apply to the whole interface. I also tried to set a dark theme in Ubuntu, many of my apps switched to a dark theme, Manuskript did not. I really wish there was a dark theme because I use the black and green 'oldschool' theme for fullscreen editing, and switching back to the bright UI can be quite jarring.

This seems to be as close as I can get to an actual dark theme
![image](https://user-images.githubusercontent.com/38058764/62912358-ccf52b00-bd55-11e9-8bed-49cded1d8546.png)

**Comment by gedakc** [comment]
Added [How to Enable Dark Theme](https://github.com/olivierkes/manuskript/wiki/How-to-Enable-Dark-Theme) wiki entry.

**Comment by corruleum** [comment]
I also do not have a dark theme available in Manuskript, similar to zja203. Linux Mint 18.2, MATE, near as I can tell it uses GTK (GTK3?) for my system dark theme (not KDE Breeze Dark). All (but one) other programs I use either respect that or provide their own dark theme.

The two theme options I have available for Manuskript theme at all: Fusion (the default, and a light theme), or Windows. Neither of these appear to pay any attention to my system's dark theme.

**Comment by zeth** [comment]
A rather belated reply to @zja203, the issue is not Manuskript, it is Ubuntu not sending your theme options across to QT apps. To solve this, assuming stock Ubuntu 19.10:

sudo apt qt5-style-plugins qt5ct

Then using the qt5ct application choose the gtk2 option and set it up as you want. (Not sure if this bit is needed).

Then in Manuskript's settings, you will see you now have more than the two original options, choose gtk2. As shown here:
  
![settings](https://user-images.githubusercontent.com/945574/79636085-1f9ca780-816d-11ea-9c20-1a5fdd172c7f.png)

**Comment by dmjohnsson23** [comment]
I just added some instructions that worked for me to get dark mode for Manuskript on Ubuntu/Gnome to the wiki page. I thought I'd point out here however, that (at least in my case) markdown headings in the text are unreadable in dark mode due to the dark blue font color used. I wasn't able to find a way to change that.

**Comment by zeth** [comment]
Interesting, using the gtk2 theme you get them in a kind of pink/salmon colour.
![salmon](https://user-images.githubusercontent.com/945574/79686911-a3bd6080-823b-11ea-9d90-7d06b8a96fec.png)

**Comment by zeth** [comment]
When I use the adwaita-qt, they seem readable on my screen in blue, although I seem to get other problems.

![adwaita](https://user-images.githubusercontent.com/945574/79687015-45dd4880-823c-11ea-801b-ae8e9a7b718e.png)

**Comment by zeth** [comment]
For me on adwaita-qt, the buttons go into extreme geek mode with no labels, like here in the outline, the new chapter and delete buttons are empty. I hope I don't press the wrong one!
![wonder](https://user-images.githubusercontent.com/945574/79687101-d7e55100-823c-11ea-967e-bbc1fbd5e8bf.png)

**Comment by zeth** [comment]
Apparently this might be fixed in Ubuntu 20.04
https://github.com/ubuntu/yaru/issues/1294

**Comment by zeth** [comment]
Okay, I have tried lots and lots themes for dark mode on Ubuntu and the one that doesn't seg fault (yet) and has no major visual drawbacks is called kvantum. 

sudo apt install qt5-style-kvantum qt5-style-kvantum-themes 

Then choose kvantum in Manuskript.
![nice](https://user-images.githubusercontent.com/945574/79698446-d9d10380-8280-11ea-8760-a8f5d06d12ac.png)

**Comment by zeth** [comment]
For the themes that crash, I got out gdb and tried to figure out what is happening but I fear the problems are in QT or with the themes themselves. I couldn't find anything that the Manuskript code is doing wrong. However, I am not a QT expert so your mileage may vary. I have found one dark theme that works (so far) and so I am happy.

**Comment by majaminnaert** [comment]
Odd thing though, kvantum doesn't show the editor in night mode. whooptie.
Would be so awesome if I could load that solarized-dark theme I like to use in some of my code editing environments ^^
Meantime, tweaks seem to work in a way.

**Comment by zeth** [comment]
> Odd thing though, kvantum doesn't show the editor in night mode. whooptie.

Yes sorry, that's the way I like it. The text of my book is black on white but the program is dark. That way I can focus on my text and let the hard work of the Manuskript devs disappear into the background.

**Comment by AlanDunaway** [comment]
> A rather belated reply to @zja203, the issue is not Manuskript, it is Ubuntu not sending your theme options across to QT apps. The way I solved this, using the stock GNOME based Ubuntu 19.10:
> 
> sudo apt install qt5-style-plugins
> 
> Then in Manuskript's settings, I found that I had more than the two original options, I chose gtk2 and the application went suitably black and orange. As shown here:
> 
> ![settings](https://user-images.githubusercontent.com/945574/79636085-1f9ca780-816d-11ea-9c20-1a5fdd172c7f.png)

As soon as I select gtk, Manuskript crashes on 22.04.
